bahmnicore-omod/src/test/java/org/bahmni/module/bahmnicore/web/v1_0/controller/VisitDocumentControllerTest.java

```diff
@@ -160,4 +163,43 @@ public void shouldNotCallDeleteWithGivenFileNameIfFileNameIsEmpty() throws Excep
         ResponseEntity<Object> responseEntity = visitDocumentController.deleteDocument("");
         Assert.assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
     }
+
+    @Test
+    public void shouldFailIfFileNameWithSpecialCharsOtherThanDashAndUnderscoreIsPassedInRequest() throws Exception {
+        PowerMockito.mockStatic(Context.class);
+        PowerMockito.when(Context.getPatientService()).thenReturn(patientService);
+        Patient patient = new Patient();
+        patient.setId(1);
+        patient.setUuid("patient-uuid");
+        when(patientService.getPatientByUuid("patient-uuid")).thenReturn(patient);
+
+        Document document = new Document("abcd", "jpeg", "consultation", "patient-uuid", "image", "file/name");
+
+        visitDocumentController.saveDocument(document);
+    }
+
+    @Test
+    public void shouldSaveIfFileNameNotWithSpecialCharsOtherThanDashAndUnderscoreIsPassedInRequest() {
+        PowerMockito.mockStatic(Context.class);
+        PowerMockito.when(Context.getPatientService()).thenReturn(patientService);
+        Patient patient = new Patient();
+        patient.setId(1);
+        patient.setUuid("patient-uuid");
+        when(patientService.getPatientByUuid("patient-uuid")).thenReturn(patient);
+
+        Document document = new Document("abcd", "jpeg", "consultation", "patient-uuid", "image", "file-name");
+
+        HashMap<String, String> mapWithUrl = visitDocumentController.saveDocument(document);
+        if (mapWithUrl!=null) {
+	        String documentSavedPath = mapWithUrl.get("url");
+	        if (documentSavedPath!=null) {
+	        	assertTrue(documentSavedPath.endsWith("__file-name.jpeg"));
+	        }
+        }
+        // Old files will follow: patientid-encounterName-uuid.ext (eg. 6-Patient-Document-706a448b-3f10-11e4-adec-0800271c1b75.jpeg)
+        // New ones will follow: patientid_encounterName_uuid__filename.ext (eg. 6-Patient-Document-706a448b-3f10-11e4-adec-0800271c1b75__file-name.jpeg)
```

Comment:
This comment seems incorrect. The leftside says underscore(__) but e.g shows hyphens (-).